src/main/kotlin/no/nav/syfo/infrastructure/cronjob/PubliserForhandsvarselCronjob.kt

```diff
@@ -0,0 +1,41 @@
+package no.nav.syfo.infrastructure.cronjob
+
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.application.service.VarselService
+import org.slf4j.LoggerFactory
+
+class PubliserForhandsvarselCronjob(
+    private val varselService: VarselService,
+) : Cronjob {
+    override val initialDelayMinutes: Long = 5
+    override val intervalDelayMinutes: Long = 10
+
+    override suspend fun run() {
+        val result = runJob()
+        log.info(
+            "Completed publishing forhandsvarsel with result: {}, {}",
+            StructuredArguments.keyValue("failed", result.failed),
+            StructuredArguments.keyValue("updated", result.updated),
+        )
+    }
+
+    fun runJob(): CronjobResult {
+        val result = CronjobResult()
+        val unpublishedVarsler = varselService.getUnpublished()
+        unpublishedVarsler.forEach {
+            try {
+                varselService.publish(it)
+                result.updated++
+            } catch (e: Exception) {
+                log.error("Exception caught while publishing forhandsvarsel", e)
+                result.failed++
+            }
+        }
+
+        return result
+    }
```

Comment:
Hva med å gjøre mer av denne logikken i applikasjonslaget? Typ ha en funksjon som returnerer hvor mange den feiler og hvor mange den fullfører for? Ser ut som det eneste vi er interessert i her er et resultat på antall oppdatert og antall feilet. Lagene lenger ned bør uansett returnere det de gjorde en operasjon på som et slags resultat og ikke bare `Unit`. 
Usikker på hvilken type som best gjenspeiler dette da. Sånn ideelt sett skulle vi kanskje hatt en type som ligner på dette `Resultat<AntallFeilet, FullforteVarsler>`🤔

Hva tenker du? 😊
```
fun publishUnpublishedVarsler(): List<UnpublishedVarsel?> {
        val unpublishedVarsler = varselRepository.getUnpublishedVarsler()
        val publishedVarsler = unpublishedVarsler.map { varsel ->
            try {
                varselProducer.sendArbeidstakerVarsel(varsel)
                varselRepository.setPublished(varsel)
                varsel
            } catch (e: Exception) {
                null
            }
        }
        return publishedVarsler
    }
```

Reply:
Ja, enig i at det blir mest riktig å ha det som én "tjeneste" i applikasjonslaget. Kan prøve å refaktorere :+1:

Reply:
Endret litt nå. Ble så mange resultat-aktige typer så jeg endre opp med å lage en felles i application/model.

Reply:
Litt redd for at den `Result` typen som ligger i application/model blir for generell🤔 Også litt rart at applikasjonslaget skal implementere typer som er tilpasset infrastruktur laget. Hva med å bruke kotlin sin innebygde `Result<T>` type? 😊 
Funksjonen hadde i så fall blitt seende slik ut👇 Tenker det også leser ganske bra. Failure i result inneholder også en exception som `PubliserForhandsvarselCronjob` kan logge om vi ønsker det.
```
fun publishUnpublishedVarsler(): List<Result<UnpublishedVarsel>>
```

Reply:
Godt forslag! Kan se hvordan det blir :+1:

Reply:
Synes det ble veldig bra! 🙌